internal/commands/app/create_test.go

```diff
@@ -0,0 +1,191 @@
+
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+	"github.com/10gen/realm-cli/internal/local"
+	"github.com/10gen/realm-cli/internal/utils/test/assert"
+	"github.com/10gen/realm-cli/internal/utils/test/mock"
+
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+func TestAppCreateSetup(t *testing.T) {
+	t.Run("should construct a realm and atlas client with configured base urls", func(t *testing.T) {
+		profile := mock.NewProfile(t)
+		profile.SetRealmBaseURL("http://localhost:8080")
+		profile.SetAtlasBaseURL("http://localhost:8888")
+
+		cmd := &CommandCreate{inputs: createInputs{newAppInputs: newAppInputs{
+			Name: "test-app",
+		}}}
+		assert.Nil(t, cmd.realmClient)
+		assert.Nil(t, cmd.atlasClient)
+		assert.Nil(t, cmd.Setup(profile, nil))
+		assert.NotNil(t, cmd.realmClient)
+		assert.NotNil(t, cmd.atlasClient)
+	})
+}
+
+func TestAppCreateHandler(t *testing.T) {
+	t.Run("should create bare bone project when no from type is specified", func(t *testing.T) {
+		profile, teardown := mock.NewProfileFromTmpDir(t, "app_create_test")
+		defer teardown()
+
+		var createdApp realm.App
+		client := mock.RealmClient{}
+		client.CreateAppFn = func(groupID, name string, meta realm.AppMeta) (realm.App, error) {
+			createdApp = realm.App{
+				GroupID: groupID,
+				Name:    name,
+				AppMeta: meta,
+			}
+			return createdApp, nil
+		}
+		client.ImportFn = func(groupID, appID string, appData interface{}) error {
+			return nil
+		}
+
+		cmd := &CommandCreate{
+			inputs: createInputs{newAppInputs: newAppInputs{
+				Name:            "test-app",
+				Project:         "test-project",
+				Location:        realm.LocationVirginia,
+				DeploymentModel: realm.DeploymentModelGlobal,
+			}},
+			realmClient: client,
+		}
+
+		assert.Nil(t, cmd.Handler(profile, nil))
+
+		data, readErr := ioutil.ReadFile(filepath.Join(profile.WorkingDirectory, cmd.inputs.Name, local.FileRealmConfig.String()))
+		assert.Nil(t, readErr)
+
+		var config local.AppRealmConfigJSON
+		assert.Nil(t, json.Unmarshal(data, &config))
+		assert.Equal(t, local.AppRealmConfigJSON{local.AppDataV2{local.AppStructureV2{
+			ConfigVersion:   realm.DefaultAppConfigVersion,
+			Name:            "test-app",
+			Location:        realm.LocationVirginia,
+			DeploymentModel: realm.DeploymentModelGlobal,
+		}}}, config)
+		assert.Equal(t, realm.App{
+			GroupID: "test-project",
+			Name:    "test-app",
+			AppMeta: realm.AppMeta{
+				Location:        realm.LocationVirginia,
+				DeploymentModel: realm.DeploymentModelGlobal,
+			},
+		}, createdApp)
+	})
+
+	t.Run("should create a templated app when from is set", func(t *testing.T) {
+		profile, teardown := mock.NewProfileFromTmpDir(t, "app_create_test")
+		defer teardown()
+
+		testApp := realm.App{
+			ID:          primitive.NewObjectID().Hex(),
+			GroupID:     primitive.NewObjectID().Hex(),
+			ClientAppID: "from-app-abcde",
+			Name:        "from-app",
+		}
+
+		var zipCloser *zip.ReadCloser
+		defer func() { zipCloser.Close() }()
```

Comment:
instead of this, just have

```go
zipPkg, err := zip.OpenReader("testdata/project.zip")
assert.Nil(t, err)
```

and then in your mocked `ExportFn`, just `return ", &zipPkg.Reader, nil`